src/matcha_ml/core/core.py

```diff
@@ -24,31 +24,49 @@ def get(
         Dict[str, Dict[str, str]]: the information of the provisioned resource.
 
     Raises:
+        MatchaError: Raised when the matcha state has not been initialized
         MatchaError: Raised when the matcha.state file does not exist
         MatchaInputError: Raised when the resource or property name does not exist in the matcha.state file
     """
     matcha_state_service = MatchaStateService()
+    remote_state = RemoteStateManager()
 
-    if not matcha_state_service.state_file_exists:
```

Comment:
Nit: Again, probably beyond scope but why do we use an attribute in the local service and a method in remote?

Reply:
I don't have a good answer for this.